File: tmdb-core/src/main/java/dev/reuss/tmdb/value/id/CollectionId.java
package dev.reuss.tmdb.value.id;

/**
 * Type-safe TMDB collection id.
 *
 * @param value raw TMDB collection id
 */
public record CollectionId(int value) implements NumericTmdbResourceId {

    public CollectionId {
        if (value <= 0) {
            throw new IllegalArgumentException("Collection id must be greater than 0");
        }
    }

    public static CollectionId of(int value) {
        return new CollectionId(value);
    }
}

File: tmdb-core/src/main/kotlin/dev/reuss/tmdb/value/id/CollectionId.kt
package dev.reuss.tmdb.value.id

/**
 * Type-safe TMDB collection id.
 */
class CollectionId private constructor(
    value: Int,
) : NumericTmdbResourceId(value, "Collection id") {
    companion object {
        /**
         * Creates an identifier from a positive TMDB resource ID.
         *
         * @throws IllegalArgumentException if [value] is not greater than `0`
         */
        @JvmStatic
        fun of(value: Int): CollectionId = CollectionId(value)
    }
}
